header/index.js

```diff
@@ -1,17 +1,24 @@
+const Typeahead = require('./js/typeahead');
+const promoHandler = require('./js/promoHandler');
+
 function init (flags) {
-	require('n-header-footer/header_new').init(flags);
+	require('o-header').init();
+	promoHandler.init(flags);
 
 	if (flags.get('fancyDrawer')) {
 		require('n-header-footer/drawer').init();
 	}
 
-	if (flags.get('meganav')) {
-		require('n-header-footer/meganav').init();
-	}
-
 	if (flags.get('sectionBreadcrumbs')) {
 		require('n-header-footer/breadcrumb').init();
 	}
+
+	const typeaheadElements = document.querySelectorAll('[data-typeahead]');
+	if (flags.get('typeahead') && typeaheadElements.length) {
+		for (const element of typeaheadElements) {
+			new Typeahead( element, `//${window.location.host}/search-suggestions?q=` );
+		};
```

Comment:
are iterators for nodelists implemented by polyfill service? 

If not `for (let i = 0, len = typeaheadElements.length; i < len; i++) {}` for old school bonus points (though also perf++)